src/main/java/gregtech/api/pipenet/tile/TileEntityPipeBase.java

```diff
@@ -296,14 +266,18 @@ public void readFromNBT(@Nonnull NBTTagCompound compound) {
             this.pipeBlock = block instanceof BlockPipe ? (BlockPipe<PipeType, NodeDataType, ?>) block : null;
         }
         this.pipeType = getPipeTypeClass().getEnumConstants()[compound.getInteger("PipeType")];
-        NBTTagCompound blockedConnectionsTag = compound.getCompoundTag("BlockedConnectionsMap");
-        this.openConnectionsMap.clear();
-        for (String attachmentTypeKey : blockedConnectionsTag.getKeySet()) {
-            int attachmentType = Integer.parseInt(attachmentTypeKey);
-            int blockedConnections = blockedConnectionsTag.getInteger(attachmentTypeKey);
-            this.openConnectionsMap.put(attachmentType, blockedConnections);
+
+        if (compound.hasKey("Connections")) {
+            openConnections = compound.getInteger("Connections");
+        } else {
```

Comment:
Lets do `else if (compound.hasKey("BlockedConnectionsMap")` just in case, since it could cause an NPE if it is not present